tensorboard/webapp/metrics/store/metrics_reducers.ts

```diff
@@ -1046,8 +1046,13 @@ const reducer = createReducer(
         }
       }
 
+      const newCardMetadataMap: CardMetadataMap = {};
+      const nextCardList: string[] = [];
```

Comment:
We don't want to clear the cardList and cardMetadataMap every time fetching new time series data. With the code here you're resetting it to empty list.
Also try to think about it at the case of TB dashboard. Both tag fetch and time series data fetch will be fired and they're writing to the same state. I'm not sure if the sequence is guaranteed (please check). But at least it will not start from empty list every time.

Reply:
Changing implementation, no longer an issue